community/cypher/frontend-3.3/src/main/scala/org/neo4j/cypher/internal/frontend/v3_3/SemanticState.scala

```diff
@@ -347,12 +375,19 @@ case class SemanticState(currentScope: ScopeLocation,
   def addNotification(notification: InternalNotification): SemanticState =
     copy(notifications = notifications + notification)
 
+  def implicitContextGraph(variable: Option[String], position: InputPosition, contextGraphName: String)
+  : Either[SemanticError, SemanticState] =
+    variable match {
+      case Some(name) => implicitGraph(Variable(name)(position))
+      case None => Left(SemanticError(s"No $contextGraphName in scope", position))
```

Comment:
This'll become `No foo in scope`; might be more informative to say `No graph foo in scope`.